File: backend/src/main/java/jreader2/domain/Subscription.java
package jreader2.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Builder;
import lombok.Data;
import lombok.extern.jackson.Jacksonized;

import java.time.ZonedDateTime;

@Data
@Jacksonized
@Builder
public class Subscription {

    @JsonIgnore
    private String ownerEmail;
    private long groupId;
    private long id;
    private String url;
    private String name;
    private long rank;
    private ZonedDateTime lastUpdateDate;

}

File: frontend/src/app/domain/subscription.ts
export class Subscription {

    id: number;
    groupId: number;
    url: string;
    name: string;
    rank: number;

    constructor(groupId: number, url: string, name: string) {
        this.groupId = groupId;
        this.url = url;
        this.name = name;
    }

}
